lib/chef/exceptions.rb

```diff
@@ -97,6 +97,7 @@ class GroupIDNotFound < ArgumentError; end
     class ConflictingMembersInGroup < ArgumentError; end
     class InvalidResourceReference < RuntimeError; end
     class ResourceNotFound < RuntimeError; end
+    class ProviderNotFound < ArgumentError; end
```

Comment:
I was just creating a new class in order to have an analogue to ResourceNotFound.  Up until now, that codepath was raising `ArgumentError`, so I feel like we need it to be a subclass of that--might be overthinking it though.

I'll check on NoProviderAvailable.  I'd like the resource and provider one to be named similarly.